### PR TITLE
Add inputs and if condition to `provider-ci.yml`

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR is opened to skip the report-breaking-changes job and the Report Statistics step in the check-diff job in non-upjet-based providers.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Manually tested on `turkenf/uptest` and `turkenf/provider-terraform`

### CODE REVIEW

**Review by ulucinar — APPROVED** [review]
Thanks @turkenf, lgtm.

File: .github/workflows/provider-ci.yml [hunk 1]
```diff
@@ -31,7 +36,7 @@ jobs:
   report-breaking-changes:
     runs-on: ubuntu-22.04
     needs: detect-noop
-    if: needs.detect-noop.outputs.noop != 'true'
+    if: ${{ needs.detect-noop.outputs.noop != 'true' && inputs.upjet-based-provider == true }}
```
[the remark this answers is not in the document]
Comment (ulucinar) [reply .github/workflows/provider-ci.yml:39]: We can also consider moving the condition only to the schemadiff step (we will also need to add the `crddiff` make target to provider-terraform.

[the remark this answers is not in the document]
Comment (turkenf) [reply .github/workflows/provider-ci.yml]: Moved the condition to the `Report native schema version changes` step.

File: .github/workflows/provider-ci.yml [hunk 2]
```diff
@@ -152,6 +157,7 @@ jobs:
         run: git diff
 
       - name: Report Statistics
+        if: ${{ inputs.upjet-based-provider == true }}
```
[the remark this answers is not in the document]
Comment (turkenf) [reply .github/workflows/provider-ci.yml]: I tested this suggestion and made the change.

**Pull request merged**